### ISSUE TITLE
Cache does not expire in given duration

### ISSUE THREAD

**Issue by caperneoignis** [PHPSocialNetwork/phpfastcache#266]
While doing PHPUnit testing for a class using phpfastcache, I noticed the cache does not expire after a given duration. 

````javascript
class CacheTest extends PHPUnit_Framework_TestCase
{
    private $database_;
    /**
     * @covers cache creation.
     */
    public function testCacheCreation(){
        $cache = new FastCache();
        $key = "test1";
        $duration = 1;
        //try and retrive a store, should come back as null.
        $item = $cache->RetrieveCache($key);
        $this->assertTrue(is_null($item), "the item should come back as false.");
        
        //now store something and should come back as true.
        $storeItem = "I'm a stored String";
        $cache->StoreCache($storeItem, $key,$duration);
        $item = $cache->RetrieveCache($key);
        $this->assertTrue($item == "I'm a stored String", "check something was returned");
        
        sleep(2); //go to sleep for several seconds and see if it dissappears.
        $item = $cache->RetrieveCache($key);
         //**this is coming back as false, it is returning the string ******
         $this->assertTrue(is_null($item),"Cache should be deleted after duration");
        //clear out the cache.
        $cache->clear($key);
    }
    public function testCacheDisappearing(){
        $cache = new FastCache();
        $key = "test2";
        $duration = 10;
        //try and retrive a store, should come back as null.
        $item = $cache->RetrieveCache($key);
        $this->assertTrue(is_null($item), "There should be no returned cache");
    
        //now store something and should come back as true.
        $storeItem = "I'm a stored String";
        $cache->StoreCache($storeItem, $key,$duration);
        for($i = 0; $i < 10; $i++ ){
            $item = $cache->RetrieveCache($key);
            $this->assertTrue($item == "I'm a stored String", "check something was returned");
            //sleep one second per rotation
            sleep(2);
        }
    
        //now lets check to make sure it gets deleted even if it is being looked up
        $item = $cache->RetrieveCache($key);
        //**this is coming back as false, it is returning the string ******
        $this->assertTrue(is_null($item),"try and retrieve item after duration");
        $cache->clear($key);
    }
}
````

The above is the unit testing the below is the cache class I made using phpFastCache.

````javascript
class FastCache {
    /**
     * 
     * @var $cache_ is used for storing the CacheManager implentation.
     */
    private $cache_;
    
    /**
     * default constructor
     *
     */
    public function __construct(){
        $this->cache_ = CacheManager::Files();
    }
    
    /**
     * Delete the cache by key value. Normally used for testing, but I can see a force referesh also. 
     * @param string $key
     */
    public function clear($key){
        $this->cache_->delete($key);
        $this->cache->clean();
    }
    /**
     * if the cache is retrievable then pull the information otherwise it will return null
     * not a fan of multipule returns 
     *  Will probably change this later.. 
     * @param string $key
     * @return object | null
     */
    public function RetrieveCache($key){
        $retrievedItems = $this->cache_->get($key);
        if(!is_null($retrievedItems)){
            return $retrievedItems;
        }
        else
            return null;
    }
    
    /**
     * Store the information using a key, the first item is the information to be stored
     * the next item is the key to store by. Cache is good for 30 seconds or set value.
     * @param VALUE | PRODUCTS | ARRAY | OBJECT $itemsToBeStored
     * @param string $key
     * @param int $duration = 30 (seconds)
     */
    public function StoreCache($itemsToBeStored, $key, $duration = 30){
        $this->cache_->set($key,$itemsToBeStored,$duration);
    }
}
````

The value is still getting returned even if I set the duration to 1 second. I need the duration to be a fixed size, in order to have reports refresh within a specified time frame, usually around one minute. However, the unit test are coming back with the stored values after I set a sleep timer greater then the duration. Am I doing something wrong, or did I stumble on to something?

**Comment by khoaofgod** [comment]
@Geolim4 can you help on this one. I'm not good at this.

**Comment by Geolim4** [comment]
@caperneoignis 

Hi thanks you for your report, i have some questions:

1- What's the driver tested ?
2- Can you test for a cache duration that take at least 1 minute please ?

Thanks you :)

**Comment by khoaofgod** [comment]
I think he used ::Files()  and expired time is 1 second.

**Comment by caperneoignis** [comment]
Okay, I'll extend out the test duration to a minute. And yes, I was doing files, it is set in the constructor for the FastCache class.

**Comment by caperneoignis** [comment]
Okay, I changed the duration to 63, and the corresponding sleep times. I received the same results, the last two assertTrue(s) failed. Which means the cache was not cleared after 63 seconds.

**Comment by khoaofgod** [comment]
@caperneoignis Can you use the last commit ( not release, and try again ). Let us know. We debugged but I didn't see any issues.

**Comment by khoaofgod** [comment]
Because sometime phpFastCache will try to store "all small varibles" to memory and didn't write to Files yet. Until the script finish, it will write everything to files 1 time to make the respone time / reduce hard disk access time. So, if it happened, because the sleep() there. 

On the latest comit, I disabled that behavier by default to save more memory, and only enable if user config set to TRUE.

**Comment by caperneoignis** [comment]
Okay, it wont be till Monday when I get back in the office, but I'll let you know as soon as I can try it again. Is there a better way to test? If the script doesn't write till the end of the script, can I do a phpUnitTest to see if the config exist after a minute or so after running the last phpunit test? IE just run the test then wait 2 minutes and then rerun the unit test to see if the cache did indeed clear? But I'll pull the newest commit and let you know Monday what I find. Thanks again to you all for your help in this.

**Comment by caperneoignis** [comment]
Okay, so after downloading the newest dev version, using composer. I re-ran the test. I changed a couple of things about the test. 
1. I got ride of the cache->clear command.
2. I used a stop watch to count to 63 seconds after the test case ran. 
3. I also did the same test but only after 10 seconds. 

Number 2. item passed, the cache no longer existed after a minute and 3 seconds.
Number 3. item failed as expected. I really ran that one to ensure the caches were not being cleared by code by accident. 

Thank you for your help.
